rust/connlib/tunnel/src/device_channel/tun_linux.rs

```diff
@@ -136,63 +137,24 @@ impl Tun {
             worker: Some(
                 set_iface_config(config.clone(), dns_config, handle, dns_control_method).boxed(),
             ),
+            routes: Vec::new(),
         })
     }
 
-    pub fn add_route(&mut self, route: IpNetwork, _: &impl Callbacks) -> Result<Option<Self>> {
-        let handle = self.handle.clone();
-
-        let add_route_worker = async move {
-            let index = handle
-                .link()
-                .get()
-                .match_name(IFACE_NAME.to_string())
-                .execute()
-                .try_next()
-                .await?
-                .ok_or(Error::NoIface)?
-                .header
-                .index;
-
-            let res = match route {
-                IpNetwork::V4(ipnet) => make_route_v4(index, &handle, ipnet).execute().await,
-                IpNetwork::V6(ipnet) => make_route_v6(index, &handle, ipnet).execute().await,
-            };
-
-            match res {
-                Ok(_) => Ok(()),
-                Err(NetlinkError(err)) if err.raw_code() == FILE_ALREADY_EXISTS => Ok(()),
-                // TODO: we should be able to surface this error and handle it depending on
-                // if any of the added routes succeeded.
-                Err(err) => {
-                    tracing::error!(%route, "failed to add route: {err:#?}");
-                    Ok(())
-                }
-            }
-        };
-
-        match self.worker.take() {
-            None => self.worker = Some(add_route_worker.boxed()),
-            Some(current_worker) => {
-                self.worker = Some(
-                    async move {
-                        current_worker.await?;
-                        add_route_worker.await?;
-
-                        Ok(())
-                    }
-                    .boxed(),
-                )
-            }
+    pub fn set_routes(
+        &mut self,
+        routes: Vec<IpNetwork>,
+        _: &impl Callbacks,
+    ) -> Result<Option<Self>> {
+        if routes == self.routes {
```

Comment:
I think this will also check the order so you might want to sort them first or use a `HashSet`.